go/vt/vtgate/evalengine/api_type_aggregation.go

```diff
@@ -128,6 +153,23 @@ func (ta *typeAggregation) add(tt sqltypes.Type, f typeFlag) {
 	ta.total++
 }
 
+func nextSignedTypeForUnsigned(t sqltypes.Type) sqltypes.Type {
+	switch t {
+	case sqltypes.Uint8:
+		return sqltypes.Int16
+	case sqltypes.Uint16:
+		return sqltypes.Int24
+	case sqltypes.Uint24:
+		return sqltypes.Int32
+	case sqltypes.Uint32:
+		return sqltypes.Int64
+	case sqltypes.Uint64:
+		return sqltypes.Decimal
```

Comment:
Anything where this happens in practice? Wouldn’t instead MySQL fail with an overflow error if this happens?

Reply:
No, this is not an overflow, this is explicitly listed in the refman:

> If there is a combination of signed and unsigned integer types, the result is signed and the precision may be higher. For example, if the types are signed [INT](https://dev.mysql.com/doc/refman/8.0/en/integer-types.html) and unsigned [INT](https://dev.mysql.com/doc/refman/8.0/en/integer-types.html), the result is signed [BIGINT](https://dev.mysql.com/doc/refman/8.0/en/integer-types.html).

> The exception is unsigned [BIGINT](https://dev.mysql.com/doc/refman/8.0/en/integer-types.html) combined with any signed integer type. The result is [DECIMAL](https://dev.mysql.com/doc/refman/8.0/en/fixed-point-types.html) with sufficient precision and scale 0.